Experiment_Launcher_code/experimentgui.py

```diff
@@ -177,25 +157,35 @@ def get_settings(self):
             first_opponent_strategy = self.first_opponent_type.get()
             second_opponent_strategy = self.second_opponent_type.get()
 
-            settings = {
-                'experiment_name' : self.experiment_name.get(),
-                'comport_name' : self.comport_name.get(),
-                'num_trials': num_trials,
-                'return_time': return_time,
-                'decision_time': decision_time,
-
-                'opponent_type': self.get_opponent_type(opponent_type),
-                'opponent1_strategy': first_opponent_strategy,
-                'opponent2_strategy': second_opponent_strategy,
-                'opponent1_probability' : self.first_opponent_prob.get(),
-                'opponent2_probability' : self.second_opponent_prob.get()
-            }
-
-            #print(settings)
-            return settings
 
         except ValueError:
             # Handle the case where the input is not a valid integer
             messagebox.showerror("Invalid Input", "Please enter valid numbers for trials, duration, and decision time.")
             return None
 
+    def is_data_valid(self):
```

Comment:
seems more logical to check the trial validity in the experiment gui itself instead of in the experiment launcher to me.please can you explain this change

Reply:
Fixed. Please look at the new code version.